pkg/engine/update.go

```diff
@@ -131,6 +134,22 @@ func newUpdateSource(
 		logging.V(7).Infof("newUpdateSource(): failed to install missing plugins: %v", err)
 	}
 
+	// Collect the version information for default providers.
+	defaultProviderVersions := computeDefaultProviderPlugins(languagePlugins, allPlugins)
```

Comment:
Can you just help me understand what's going on with this refactoring?  I would expect we would not need to make any changes to how the default plugin providers are installed as part of this work, since we should never be loading them?

Reply:
I'm guessing that this just made things simpler, even though we do end up computing default provider plugins in the query case.  Not pushing back on making the code clearer, just trying to understand if it was needed for correctness or not.

Reply:
Actually we _don't_ end up computing default provider plugins for the query case. At least if I understand your point.

This is not a correctness change -- luke asked me to simplify this so that the plugin loading code is shared between the query and update modes. I had to move this out to the update path so that we didn't compute this for the query case.